CMakeLists.txt

```diff
@@ -4,165 +4,163 @@
-set(CMAKE_CXX_STANDARD_REQUIRED yes CACHE BOOL "Default value for CXX_STANDARD_REQUIRED property of targets.")
-set(CMAKE_CXX_EXTENSIONS no CACHE BOOL "Default value for CXX_EXTENSIONS property of targets.")
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD
+      20
+      CACHE STRING "Default value for CXX_STANDARD property of targets."
+  )
+  option(CMAKE_CXX_STANDARD_REQUIRED "Default value for CXX_STANDARD_REQUIRED property of targets." YES)
+  option(CMAKE_CXX_EXTENSIONS "Default value for CXX_EXTENSIONS property of targets." NO)
+endif()
+
 option(BOOST_UT_ENABLE_MEMCHECK "Run the unit tests and examples under valgrind if it is found" OFF)
 option(BOOST_UT_ENABLE_COVERAGE "Run coverage" OFF)
 option(BOOST_UT_ENABLE_SANITIZERS "Run static analysis" OFF)
-option(BOOST_UT_ENABLE_RUN_AFTER_BUILD "Automatically run built artifacts. If disabled, the tests can be run with ctest instead" ON)
 option(BOOST_UT_BUILD_BENCHMARKS "Build the benchmarks" ${MASTER_PROJECT})
 option(BOOST_UT_BUILD_EXAMPLES "Build the examples" ${MASTER_PROJECT})
 option(BOOST_UT_BUILD_TESTS "Build the tests" ${MASTER_PROJECT})
 
-add_library(boost.ut INTERFACE)
-target_include_directories(boost.ut INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
-add_library(boost::ut ALIAS boost.ut)
-
-if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-  if (WIN32) # clang-cl
-    target_compile_options(boost.ut INTERFACE
-                        -Weverything
-                        -Werror
-                        -pedantic
-                        -pedantic-errors
-                        -Wno-c++98-compat
-                        -Wno-c++98-compat-pedantic
-                        -Wno-c99-extensions)
+# ---- Add dependencies via CPM ----
+# see https://github.com/cpm-cmake/CPM.cmake for more info
+
+if(CMAKE_VERSION VERSION_LESS 3.20.0)
+  # see https://github.com/TheLartians/PackageProject.cmake/pull/19
+  include(cmake/PackageProject.cmake)
+else()
+  include(cmake/CPM.cmake)
+
+  # PackageProject.cmake will be used to make our target installable
+  CPMAddPackage(
+    NAME PackageProject.cmake
+    GITHUB_REPOSITORY TheLartians/PackageProject.cmake
+    VERSION 1.6.0
+  )
+endif()
+
+add_library(ut INTERFACE)
+
+#
+# TODO: IMHO NO! add_library(boost.ut ALIAS ut) CK
+#
+# /usr/local/include/boost -> ../Cellar/boost/1.75.0_1/include/boost
+# TODO: set(INCLUDE_INSTALL_DIR boost-${BOOST_VERSION}/include)
+
+set(INCLUDE_INSTALL_DIR include/${PROJECT_NAME}-${PROJECT_VERSION}/include)
+# XXX variant: set(INCLUDE_INSTALL_DIR include)
+target_include_directories(ut INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>)
+if(APPLE)
+  target_compile_features(ut INTERFACE cxx_std_20)
+  # else() target_compile_features(ut INTERFACE cxx_std_17)
+endif()
+
+if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+  if(WIN32) # clang-cl
+    # FIXME: we should not export this pedantic options! CK
```

Comment:
clang-cl is buggy for me?

Reply:
not sure :thinking:

Reply:
hundreds of warnings is not normal?

Reply:
I hope not

Reply:
see https://github.com/ClausKlein/ut/runs/1981508898?check_suite_focus=true